src/Microsoft.ML.LightGBM/WrappedLightGbmBooster.cs

```diff
@@ -96,10 +96,10 @@ internal unsafe string GetModelString()
 
         private static double[] Str2DoubleArray(string str, char delimiter)
         {
-            return str.Split(delimiter).Select(
-                x => { double t; double.TryParse(x, out t); return t; }
-            ).ToArray();
-
+            return str.Split(delimiter)
+                      .Select(s => double.TryParse(s.Replace("inf", "∞"), out double rslt) ? rslt :
```

Comment:
.Replace and .Contains are case sensitive - should these be insensitive?
I know the old code did not do this - but do we need to check for empty string? Or is it assumed that check has been done before this call?

Reply:
`Select` could filter out empty cases, I believe. That strings are LightGBM format, so let's leave them as they are.

Reply:
Why is this replacing `inf` with `∞` rather than with the culture-insensitive string (which is `Infinity`)

Reply:
What's the difference?

---
In reply to: [243438498](https://github.com/dotnet/machinelearning/pull/1934#discussion_r243438498) [](ancestors = 243438498)

Reply:
`∞` will only parse successfully for cultures where `NumberFormatInfo.PositiveInfinitySymbol == "∞"`; the latter is the invariant case and is stable/won't change. You should probably be doing: `double.TryParse(value, out double result, CultureInfo.InvariantCulture)` for anything that you aren't prepared to support culture specific inputs on.

Reply:
Also, for this particular case, it is probably better to just have the following (or something very similar, depending on the exact inputs expected):
```csharp
string sTrim = s.Trim();
if (sTrim.Equals("inf", StringComparison.OrdinalIgnoreCase))
{
    return double.PositiveInfinity;
}
else if (sTrim.Equals("-inf", StringComparison.OrdinalIgnoreCase))
{
    return double.NegativeInfinity;
}
```

`double.Parse` and `double.TryParse` require that the entire string match, so doing a `string.Replace` is relatively pointless and just incurs additional allocations/overhead